packages/editor/src/systems/ClickPlacementSystem.tsx

```diff
@@ -0,0 +1,280 @@
+
+  useEffect(() => {
+    if (!placementModel) return
+    const sceneID = getModelSceneID(props.placementEntity)
+    if (!sceneState.scenes[sceneID]) return
+    iterateEntityNode(props.placementEntity, (entity) => {
+      const mesh = getOptionalComponent(entity, MeshComponent)
+      if (!mesh) return
+      const material = mesh.material as Material
+      clickState.materialCache.set((prev) => [...prev, [mesh, material]])
+      mesh.material = new HolographicMaterial({})
+    })
+  }, [placementModel?.scene, sceneState.scenes.keys])
+
+  return null
+}
+
+const objectLayerQuery = defineQuery([ObjectLayerComponents[ObjectLayers.Scene]])
+
+export const ClickPlacementSystem = defineSystem({
+  uuid: 'ee.studio.ClickPlacementSystem',
+  insert: { before: TransformSystem },
+  reactor: () => {
+    const clickState = useState(getMutableState(ClickPlacementState))
+    const editorState = useState(getMutableState(EditorHelperState))
+    const sceneState = useState(getMutableState(SceneState))
+
+    const renderers = defineQuery([RendererComponent])
+
+    const getParentEntity = () => {
+      if (!sceneState.scenes.value) return null
+      const sceneID = sceneState.scenes.keys[0] as SceneID
+      const scene = SceneState.getScene(sceneID)
+      if (!scene) return null
+      const entity = UUIDComponent.getEntityByUUID(scene.scene.root)
+      return entity
+    }
+
+    const createPlacementEntity = (parentEntity: Entity) => {
+      const placementEntity = createSceneEntity('Placement', parentEntity)
+      //removeComponent(placementEntity, SourceComponent)
+      return placementEntity
+    }
+
+    const clickListener = () => {
+      if (!clickState.selectedAsset.value) return
+      const parentEntity = getParentEntity()
+      if (!parentEntity) return
+      const placementEntity = clickState.placementEntity.value
+      if (!placementEntity) return
+      const sceneID = getComponent(parentEntity, SourceComponent)
+      setComponent(placementEntity, SourceComponent, sceneID)
+      setComponent(placementEntity, EntityTreeComponent, { parentEntity })
+      const snapshot = SceneSnapshotState.cloneCurrentSnapshot(sceneID)
+      const uuid = getComponent(placementEntity, UUIDComponent)
+      snapshot.data.entities[uuid] = toEntityJson(placementEntity)
+
+      dispatchAction(SceneSnapshotAction.createSnapshot(snapshot))
+
+      clickState.placedEntity.set(placementEntity)
+      clickState.placementEntity.set(createPlacementEntity(parentEntity))
+      for (const [mesh, material] of clickState.materialCache.value) {
+        mesh.material = material
+      }
+      clickState.materialCache.set([])
+    }
+
+    useEffect(() => {
+      const placementMode = editorState.placementMode.value
+      const renderer = getComponent(renderers()[0], RendererComponent)
+      const canvas = renderer.canvas
+      if (placementMode === PlacementMode.CLICK) {
```

Comment:
use spatial input instead of event listeners, also, this does not clean up the click listener as a different instance of the function will be created each time the reactor rerenders